framework/src/problems/FEProblemBase.C

```diff
@@ -4283,6 +4288,9 @@ FEProblemBase::checkExceptionAndStopSolve()
 {
   TIME_SECTION(_check_exception_and_stop_solve_timer);
 
+  if (_skip_exception_check)
```

Comment:
I would put this test above the timer. No need to create and destroy a timer in the case where you aren't going to do the work.